cmd/bucket-versioning-handler.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * MinIO Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/xml"
+	"io"
+	"net/http"
+
+	humanize "github.com/dustin/go-humanize"
+	"github.com/gorilla/mux"
+	xhttp "github.com/minio/minio/cmd/http"
+	"github.com/minio/minio/cmd/logger"
+	"github.com/minio/minio/pkg/bucket/policy"
+	"github.com/minio/minio/pkg/bucket/versioning"
+)
+
+const (
+	emptyVersioningResponse = `<VersioningConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/"/>`
+	bucketVersioningConfig  = "versioning.xml"
+
+	// Maximum size of default bucket versioning configuration allowed
+	maxBucketVersioningConfigSize = 1 * humanize.MiByte
+)
+
+// PutBucketVersioningHandler - PUT Bucket Versioning.
+// ----------
+func (api objectAPIHandlers) PutBucketVersioningHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "PutBucketVersioning")
+
+	defer logger.AuditLog(w, r, "PutBucketVersioning", mustGetClaimsFromToken(r))
+
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
```

Comment:
```suggestion
	bucket := vars["bucket"]
	if err := s3utils.CheckValidBucketName(bucket); err != nil {
		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrInvalidBucketName), r.URL, guessIsBrowserReq(r))
		return
	}	
```
That way to reject early and don't have to rely on `objectAPI` implementations.

Reply:
Yeah we can do this way before in all calls, we need to refactor this outside of versioning changes.

Reply:
I suppose this can only be done to existing buckets anyway.